pkg/vpcmodel/semanticDiff.go

```diff
@@ -271,6 +259,58 @@ func (connDiff *connectivityDiff) string(diffAnalysis diffAnalysisType, thisMinu
 	return res
 }
 
+// connDiffDecode decode connectivesDiff information for grouping:
+// this includes the following two strings separated by ";"
+//  1. diff-type info: e.g. diff-type: removed
+//  2. configs info and info regarding missing endpoints:
+//     e.g.: config1: All Connections, config2: No connection, vsis-diff-info: vsi0 removed
+func connDiffDecode(src, dst VPCResourceIntf, connDiff *connectionDiff,
+	diffAnalysis diffAnalysisType, thisMinusOther bool) string {
+	conn1Str, conn2Str := conn1And2Str(connDiff, thisMinusOther)
+	diffType, endpointsDiff := diffAndEndpointsDisc(connDiff.diff, src, dst, thisMinusOther)
+	diffInfo := diffInfoStr(diffAnalysis)
+	diffTypeStr := fmt.Sprintf("%v %s", diffTypeStr, diffType)
+	connDiffStr := fmt.Sprintf(configsStr, conn1Str, conn2Str, diffInfo, endpointsDiff)
+	return diffTypeStr + semicolon + connDiffStr
+}
+
+func diffInfoStr(diffAnalysis diffAnalysisType) string {
+	if diffAnalysis == Subnets {
+		return "subnets-diff-info:"
+	} else if diffAnalysis == Vsis {
+		return "vsis-diff-info:"
+	}
+	return ""
+}
+
+func conn1And2Str(connDiff *connectionDiff, thisMinusOther bool) (conn1Str, conn2Str string) {
+	if thisMinusOther {
+		conn1Str = connStr(connDiff.conn1)
+		conn2Str = connStr(connDiff.conn2)
+	} else {
+		conn1Str = connStr(connDiff.conn2)
+		conn2Str = connStr(connDiff.conn1)
+	}
+	return conn1Str, conn2Str
+}
+
+// connDiffEncode decode the above string
+func connDiffEncode(src, dst EndpointElem, decoded string) string {
+	encoded := strings.Split(decoded, semicolon)
+	printDiff := fmt.Sprintf("%s, source: %s, destination: %s, %s\n", encoded[0], src.Name(), dst.Name(), encoded[1])
+	return printDiff
```

Comment:
what is `decoded`? what is `encoded[0]` , `encoded[1]` ?
can you avoid relying on strings formats and splitting strings? it is usually error-prone and more difficult to maintain the code, as printed formats may change..

Reply:
1. We rely on string formats in other cases in grouping as well, by using `func (conn *ConnectionSet) EnhancedString() string`. Keeping the format of the grouping as is is beneficial, and I thing its better to keep it as is. Namely, to encode all info other than `src` and `dst` into a string for grouping. 
2. However, I agree that the decoding should be done more carefully and specifically it should be reused, e.g. by other outputs format that I begun writing code for. So I will change this function so it returns the different semantic components instead of a single string. This way we will have a single encoding func and a single decoding func. This is not too error prone, I believe
3. It also seems that I switched the decode and encode terms. Will fix.

Reply:
done